pipeline/report.py

```diff
@@ -244,136 +429,91 @@ def make(self, key):
-                ipsi_cdend_2 = proj_contra_trial_g2[:, np.logical_and(
-                    time_stamps >= p_start, time_stamps < p_end)].mean(axis=1)
+            for v in probe_tracks_list:
+                ax.plot(v[:, 0], v[:, 2], v[:, 1], 'r')
 
-            c_df = pd.DataFrame([contra_cdend_1, contra_cdend_2]).T
-            c_df.columns = labels
-            c_df['trial-type'] = 'contra'
-            i_df = pd.DataFrame([ipsi_cdend_1, ipsi_cdend_2]).T
-            i_df.columns = labels
-            i_df['trial-type'] = 'ipsi'
-            df = c_df.append(i_df)
-
-            # remove NaN trial - could be due to some trials having no spikes
-            non_nan = ~np.logical_or(np.isnan(df[labels[0]]).values, np.isnan(df[labels[1]]).values)
-            df = df[non_nan]
-
-            fig = plt.figure(figsize=(6, 6))
-            jplot = jointplot_w_hue(data=df, x=labels[0], y=labels[1], hue='trial-type', colormap=['b', 'r'],
-                                    figsize=(8, 6), fig=fig, scatter_kws=None)
-
-            # ---- plot this fig into the main figure ----
-            buf = io.BytesIO()
-            fig.savefig(buf, format='png')
-            buf.seek(0)
-            axs[p1, p2].imshow(Image.open(buf))
-            buf.close()
-            plt.close(fig)
+            ax.set_title('Probe Track in CCF (um)')
 
         # ---- Save fig and insert ----
-        fn_prefix = f'{water_res_num}_{sess_date}_'
+        fn_prefix = (experiment.Project & key).fetch1('project_name') + '_'
 
         fig_dict = {}
-        for fig, figname in zip((fig1,), ('coding_direction',)):
-            fig_fp = sess_dir / (fn_prefix + figname + '.png')
+        for fig, figname in zip((fig1,), ('tracks_plot',)):
+            fig_fp = proj_dir / (fn_prefix + figname + '.png')
             fig.tight_layout()
             fig.savefig(fig_fp)
             print(f'Generated {fig_fp}')
             fig_dict[figname] = fig_fp.as_posix()
 
         plt.close('all')
-        self.insert1({**key, **fig_dict})
+        self.insert1({**key, **fig_dict, 'session_count': session_count})
 
 
 # ---------- HELPER FUNCTIONS --------------
+
+
 def get_wr_sessdate(key):
     water_res_num, sess_date = (lab.WaterRestriction * experiment.Session & key).fetch1(
         'water_restriction_number', 'session_date')
     return water_res_num, datetime.strftime(sess_date, '%Y%m%d')
 
+
+def delete_outdated_probe_tracks(project_name='MAP'):
+    if {'project_name': project_name} not in ProjectLevelProbeTrack:
+        return
+
+    sess_count = (ProjectLevelProbeTrack & {'project_name': project_name}).fetch1('session_count')
+    latest_sess_count = len(SessionLevelProbeTrack())
+
+    if sess_count != latest_sess_count:
+        uuid_byte = (ProjectLevelProbeTrack & {'project_name': project_name}).proj(ub='(tracks_plot)').fetch1('ub')
+        ext_key = {'hash': uuid.UUID(bytes=uuid_byte)}
+
+        with ProjectLevelProbeTrack.connection.transaction:
```

Comment:
I think we should have a code comment or issue that the 'special delete' here should be revisited once datajoint filepath has this capability